modules/triggers/create-trigger.js

```diff
@@ -3,47 +3,60 @@
  * @property {string} name - name of the trigger
  * @property {string[]} aliases - alternative triggers
  * @property {?string[]} triggers - combined array of name + aliases
- * @property {function} action - what to do when the trigger is fired
- * @property {?string} permission - the permission
+ * @property {function} runner - what to do when the trigger is fired
+ * @property {?string[]} permissions - the permission(s) require to execute the trigger.
  * @property {?string[]} helpList - array of triggers to add to the help list
  */
 
 /**
  * Creates and returns a new Trigger
  * @param {string} name
- * @param {function} action
+ * @param {function} runner
  * @param {string[]} [aliases]
- * @param {string|null} [permission=null]
+ * @param {string[]|null} [permissions=null]
  * @param {?string[]|null} [addToHelpList]
  * @returns {Trigger}
  */
 const createTrigger = (
   name,
-  action,
+  runner,
   aliases,
-  permission = null,
+  permissions = null,
   addToHelpList
 ) => {
   if (!name) {
     throw new Error('A name is required for triggers');
   }
 
-  if (action === undefined) {
+  if (runner === undefined) {
     throw new Error(`Trigger "${name}" action must have an action`);
   }
 
-  if (typeof action !== 'function') {
+  if (typeof runner !== 'function') {
     throw new Error(`Trigger "${name}" action must be a function`);
   }
 
   const triggers = [name, ...aliases];
 
+  const hasPermissions = message => {
+    if (!permissions) return true;
+    for (const perm of permissions) {

```

Comment:
I would use `Array.some()` here instead of a for loop e.g.

```js
return permissions.some(perm => message.member.hasPermission(perm))
```

Reply:
Updated!